certbot-nginx/certbot_nginx/_internal/configurator.py

```diff
@@ -905,19 +927,23 @@ def _enable_redirect_single(self, domain, vhost):
             logger.info("Redirecting all traffic on port %s to ssl in %s",
                 self.DEFAULT_LISTEN_PORT, vhost.filep)
 
-    def _enable_ocsp_stapling(self, domain, chain_path):
+    def _enable_ocsp_stapling(self, domain: str, chain_path: Union[str, List[str], None]) -> None:
```

Comment:
`Union[str, List[str], None]`: Just asking a question, because I'm not an expert in typing, but isn't this a case for `Optional[str, List[str]]`

Reply:
You mean `Optional[Union[str, List[str]]]`. Well, technically a `Optional[T]` is a `Union[None, T]` in fact. Most of the time `Optional` is more elegant, but if we are talking about a `Union` type, I find a little more elegant to use only two levels of generics (here `Union` and `List`) instead of three (`Union`, `List`  and `Optional`).

For sure PEP 604 is way more elegant, because we could use `List[str] | str | None`, but it is native to Python 3.10. For the other Python versions, we need `typing_extensions` module and declare `from __future__ import annotations`. I am not that in a hurry to go back to our lovely `from __future__ import absolute_import, print_function` of the era of Python 2->3 migration ^^

Reply:
Buuuut I am saying that and used `Optional[Union[str, List[str]]]` in `enhance()`... So I will be consistent and use `Optional[Union[str, List[str]]]` in `_enable_ocsp_stapling()`.